### PR TITLE
Configure cache dir

### PR DESCRIPTION
**Pull request by devinrsmith**
We should not be writing to a container's `/tmp` directory in docker, unless it's a docker volume.

Our docker images declare a `/cache` volume, so we should:

1. Set `-Djava.io.tmpdir=/cache/tmp` so that any java code using temporary files will inherit a docker volume backed tmp space
2. Introduce `-Ddeephaven.cache.dir` so that those who want more explicit "cache" space (instead of "tmp" space), can take advantage of it. (Only `AbstractScriptSession` atm, but others could use it in the future.)

### CODE REVIEW

File: docker/server-slim/src/main/docker/image-bootstrap.properties [hunk 1]
```diff
@@ -6,4 +6,6 @@ devroot=.
 stdout.toLogBuffer=true
 stderr.toLogBuffer=true
 
-deephaven.console.type=groovy
\ No newline at end of file
+deephaven.console.type=groovy
+deephaven.cache.dir=/cache
+java.io.tmpdir=/cache/tmp
```
Comment (rcaudy) [inline docker/server-slim/src/main/docker/image-bootstrap.properties:11]: Can we make this a sub-dir of the host's `/tmp`?

File: engine/table/src/main/java/io/deephaven/engine/util/CacheDir.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import java.nio.file.Path;
+
+public final class CacheDir {
+    private static final String DEEPHAVEN_CACHE_DIR = "deephaven.cache.dir";
+    private static final String JAVA_IO_TMPDIR = "java.io.tmpdir";
+
+    private static final Path cacheDir;
+
+    static {
+        final String deephavenCacheDir = System.getProperty(DEEPHAVEN_CACHE_DIR);
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/util/CacheDir.java:12]: `Configuration.getInstance().getPropertyWithDefault`?

File: engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java [hunk 3]
```diff
@@ -61,7 +60,7 @@ protected AbstractScriptSession(@Nullable Listener changeListener, boolean isDef
         this.changeListener = changeListener;
 
         final UUID scriptCacheId = UuidCreator.getRandomBased();
-        classCacheDirectory = new File(CLASS_CACHE_LOCATION, UuidCreator.toString(scriptCacheId));
+        classCacheDirectory = CLASS_CACHE_LOCATION.resolve(UuidCreator.toString(scriptCacheId)).toFile();
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java:63]: We discussed introducing props:
INSTANCE_ID (if set, this is the ID)
INSTANCE_ID_FILE (if set and INSTANCE_ID isn't this is the file we load from or generate)
else, we randomly allocate an instance ID and be sure to clean up on exit.

File: engine/table/src/main/java/io/deephaven/engine/util/CacheDir.java (hunk 2 above)
Comment (devinrsmith) [reply engine/table/src/main/java/io/deephaven/engine/util/CacheDir.java:12]: ^ this is not an existing method. But I can use Configuration, sure.

File: docker/server-slim/src/main/docker/image-bootstrap.properties (hunk 1 above)
Comment (devinrsmith) [reply docker/server-slim/src/main/docker/image-bootstrap.properties]: Punting - https://github.com/deephaven/deephaven-core/issues/1714

File: engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java (hunk 3 above)
Comment (devinrsmith) [reply engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java:64]: Punting - https://github.com/deephaven/deephaven-core/issues/1713

File: engine/table/src/main/java/io/deephaven/engine/util/CacheDir.java (hunk 2 above)
Comment (rcaudy) [reply engine/table/src/main/java/io/deephaven/engine/util/CacheDir.java]: Sorry, method is `io.deephaven.configuration.PropertyFile#getStringWithDefault`.

**Review by rcaudy — APPROVED** [review]

**Review by rcaudy — APPROVED** [review]

**Pull request merged**